src/components/landing-page/featured-transcripts/FeaturedTranscriptClient.tsx

```diff
@@ -18,6 +19,9 @@ const FeaturedTranscriptClient = ({
     setFeatured([...featuredTranscripts].sort(() => 0.5 - Math.random()));
   };
 
+  const getSourceFromTranscript = (data: Transcript) =>
```

Comment:
I meant, we get this data on the server component before it gets to this client component.

We should always offload logic to server components when we can.
For best performance we would have to modify the functions that get "featured" and "latest" transcripts.

home page load doesn't indicate any extra processing, further optimizations should in a separate PR for complete evaluation